---
title: "R_Markdown"
author: "Christopher_Zdyrski"
date: "3/13/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#This is my R Markdown document to analyze two snp files, combine them and pull out wanted data for further analysis in ggplot2. 

###Data Summary
###Get two original files

```{r}
fang_et_al_genotypes <- read.table('https://raw.githubusercontent.com/EEOB-BioData/BCB546-Spring2021/main/assignments/UNIX_Assignment/fang_et_al_genotypes.txt', header=TRUE, sep="\t",stringsAsFactors = FALSE)

snp_position <- read.table('https://raw.githubusercontent.com/EEOB-BioData/BCB546-Spring2021/main/assignments/UNIX_Assignment/snp_position.txt', header=TRUE, sep="\t", stringsAsFactors = FALSE)

```

###Attributes of fang_et_al_genotypes.txt

```{r}
dim(fang_et_al_genotypes)
str(fang_et_al_genotypes)
object.size(fang_et_al_genotypes)
```

#By inspecting this file I learned that:
#The file is 22681376 bytes.
#There are 2783 lines.
#There are 986 columns in this file.


###Attributes of snp_position.txt

```{r}
dim(snp_position)
str(snp_position)
object.size(snp_position)
```

#By inspecting this file I learned that:
#The file size is 327392 bytes.
#There are 983 lines.
#There are 15 columns in this file.


###Data processing

#Install tidyverse package
```{r}
install.packages("tidyverse")
```

#Load tidyverse package
```{r}
library(tidyverse)
```

#Seperate out teosinte (ZMPBA, ZMPIL, and ZMPJA) and maize (ZMMIL OR ZMMLR OR ZMMMR) with grep, and also get rid of column two and three, format correctly with tibble command
```{r}

teosinte_data <- fang_et_al_genotypes[grep("ZMP", fang_et_al_genotypes$Group), ]
maize_data <- fang_et_al_genotypes[grep("ZMM", fang_et_al_genotypes$Group), ]

teosinte_data_excluding_col <- select(teosinte_data, -JG_OTU, -Group)
teosinte_data_excluding_col <- as_tibble(teosinte_data_excluding_col)

maize_data_excluding_col <- select(maize_data, -JG_OTU, -Group)
maize_data_excluding_col <- as_tibble(maize_data_excluding_col)
```

#Swap col names to rows names
```{r}
swap_teosinte_data_excluding_col <- column_to_rownames(teosinte_data_excluding_col, var="Sample_ID")

swap_maize_data_excluding_col <- column_to_rownames(maize_data_excluding_col, var="Sample_ID")
```


#Transpose both genotype files to swap the rows and columns
```{r}
transposed_teosinte_data <- t(swap_teosinte_data_excluding_col)
transposed_maize_data <- t(swap_maize_data_excluding_col)

transposed_teosinte_data <- rownames_to_column(as.data.frame(transposed_teosinte_data), var = "SNP_ID")
transposed_maize_data <- rownames_to_column(as.data.frame(transposed_maize_data), var = "SNP_ID")
```


#Crop and make sure snp file is sorted
```{r}
snp_cropped <- select(snp_position, SNP_ID, Chromosome, Position)

attach(snp_cropped)
sorted_snp_position <- snp_cropped[order(SNP_ID),]
```

#Merge files (maize/teosinte with snp_position file)
```{r}
teosinte_joined <- merge(sorted_snp_position, transposed_teosinte_data, by="SNP_ID")

teosinte_joined_2 <- filter(teosinte_joined, Chromosome != "multiple" & Chromosome != "unknown")
teosinte_ascending <- arrange(teosinte_joined_2, Position)

maize_joined <- merge(sorted_snp_position, transposed_maize_data, by="SNP_ID")
maize_joined_2 <- filter(maize_joined, Chromosome != "multiple" & Chromosome != "unknown")

maize_ascending <- arrange(maize_joined_2, Position)

```


###Teosinte grep out 10 files, one for each chromosome
```{r}
teosinte_chrom_1.txt <- teosinte_ascending[grep("1$", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_1.txt, file = paste("./Teosinte_Files/teosinte_chrom_1.txt"))
teosinte_chrom_2.txt <- teosinte_ascending[grep("2$", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_2.txt, file = paste("./Teosinte_Files/teosinte_chrom_2.txt"))
teosinte_chrom_3.txt <- teosinte_ascending[grep("3$", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_3.txt, file = paste("./Teosinte_Files/teosinte_chrom_3.txt"))
teosinte_chrom_4.txt <- teosinte_ascending[grep("4$", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_4.txt, file = paste("./Teosinte_Files/teosinte_chrom_4.txt"))
teosinte_chrom_5.txt <- teosinte_ascending[grep("5$", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_5.txt, file = paste("./Teosinte_Files/teosinte_chrom_5.txt"))
teosinte_chrom_6.txt <- teosinte_ascending[grep("6$", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_6.txt, file = paste("./Teosinte_Files/teosinte_chrom_6.txt"))
teosinte_chrom_7.txt <- teosinte_ascending[grep("7$", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_7.txt, file = paste("./Teosinte_Files/teosinte_chrom_7.txt"))
teosinte_chrom_8.txt <- teosinte_ascending[grep("8$", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_8.txt, file = paste("./Teosinte_Files/teosinte_chrom_8.txt"))
teosinte_chrom_9.txt <- teosinte_ascending[grep("9$", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_9.txt, file = paste("./Teosinte_Files/teosinte_chrom_9.txt"))
teosinte_chrom_10.txt <- teosinte_ascending[grep("10", teosinte_ascending$Chromosome), ]
write.table(teosinte_chrom_10.txt, file = paste("./Teosinte_Files/teosinte_chrom_10.txt"))

```


###Teosinte make 2 files, one with the 3 headers, and another to gsub out the "?" then cbind them together, do this for all 10 files
```{r}
teosinte_chrom_1_hyphen.txt <- arrange(teosinte_chrom_1.txt, desc(Position))
ncol(teosinte_chrom_1_hyphen.txt)
teosinte_chrom_1_hyphen.txt <- apply(teosinte_chrom_1_hyphen.txt[,4:978], 2, function(teosinte_chrom_1_hyphen.txt) gsub("\\?", "-", teosinte_chrom_1_hyphen.txt))

teosinte_chrom_1.txt_three_col <- teosinte_chrom_1.txt[1:3]
teosinte_chrom_1_hyphen.txt <- cbind(teosinte_chrom_1.txt_three_col, teosinte_chrom_1_hyphen.txt)
write.table(teosinte_chrom_1_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_1_hyphen.txt"))

```

```{r}
teosinte_chrom_2_hyphen.txt <- arrange(teosinte_chrom_2.txt, desc(Position))
ncol(teosinte_chrom_2_hyphen.txt)
teosinte_chrom_2_hyphen.txt <- apply(teosinte_chrom_2_hyphen.txt[,4:978], 2, function(teosinte_chrom_2_hyphen.txt) gsub("\\?", "-", teosinte_chrom_2_hyphen.txt))

teosinte_chrom_2.txt_three_col <- teosinte_chrom_2.txt[1:3]
teosinte_chrom_2_hyphen.txt <- cbind(teosinte_chrom_2.txt_three_col, teosinte_chrom_2_hyphen.txt)
write.table(teosinte_chrom_2_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_2_hyphen.txt"))
```

```{r}
teosinte_chrom_3_hyphen.txt <- arrange(teosinte_chrom_3.txt, desc(Position))
ncol(teosinte_chrom_3_hyphen.txt)
teosinte_chrom_3_hyphen.txt <- apply(teosinte_chrom_3_hyphen.txt[,4:978], 2, function(teosinte_chrom_3_hyphen.txt) gsub("\\?", "-", teosinte_chrom_3_hyphen.txt))

teosinte_chrom_3.txt_three_col <- teosinte_chrom_3.txt[1:3]
teosinte_chrom_3_hyphen.txt <- cbind(teosinte_chrom_3.txt_three_col, teosinte_chrom_3_hyphen.txt)
write.table(teosinte_chrom_3_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_3_hyphen.txt"))
```

```{r}
teosinte_chrom_4_hyphen.txt <- arrange(teosinte_chrom_4.txt, desc(Position))
ncol(teosinte_chrom_4_hyphen.txt)
teosinte_chrom_4_hyphen.txt <- apply(teosinte_chrom_4_hyphen.txt[,4:978], 2, function(teosinte_chrom_4_hyphen.txt) gsub("\\?", "-", teosinte_chrom_4_hyphen.txt))

teosinte_chrom_4.txt_three_col <- teosinte_chrom_4.txt[1:3]
teosinte_chrom_4_hyphen.txt <- cbind(teosinte_chrom_4.txt_three_col, teosinte_chrom_4_hyphen.txt)
write.table(teosinte_chrom_4_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_4_hyphen.txt"))
```

```{r}
teosinte_chrom_5_hyphen.txt <- arrange(teosinte_chrom_5.txt, desc(Position))
ncol(teosinte_chrom_5_hyphen.txt)
teosinte_chrom_5_hyphen.txt <- apply(teosinte_chrom_5_hyphen.txt[,4:978], 2, function(teosinte_chrom_5_hyphen.txt) gsub("\\?", "-", teosinte_chrom_5_hyphen.txt))

teosinte_chrom_5.txt_three_col <- teosinte_chrom_5.txt[1:3]
teosinte_chrom_5_hyphen.txt <- cbind(teosinte_chrom_5.txt_three_col, teosinte_chrom_5_hyphen.txt)
write.table(teosinte_chrom_5_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_5_hyphen.txt"))
```

```{r}
teosinte_chrom_6_hyphen.txt <- arrange(teosinte_chrom_6.txt, desc(Position))
ncol(teosinte_chrom_6_hyphen.txt)
teosinte_chrom_6_hyphen.txt <- apply(teosinte_chrom_6_hyphen.txt[,4:978], 2, function(teosinte_chrom_6_hyphen.txt) gsub("\\?", "-", teosinte_chrom_6_hyphen.txt))

teosinte_chrom_6.txt_three_col <- teosinte_chrom_6.txt[1:3]
teosinte_chrom_6_hyphen.txt <- cbind(teosinte_chrom_6.txt_three_col, teosinte_chrom_6_hyphen.txt)
write.table(teosinte_chrom_6_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_6_hyphen.txt"))
```

```{r}
teosinte_chrom_7_hyphen.txt <- arrange(teosinte_chrom_7.txt, desc(Position))
ncol(teosinte_chrom_7_hyphen.txt)
teosinte_chrom_7_hyphen.txt <- apply(teosinte_chrom_7_hyphen.txt[,4:978], 2, function(teosinte_chrom_7_hyphen.txt) gsub("\\?", "-", teosinte_chrom_7_hyphen.txt))

teosinte_chrom_7.txt_three_col <- teosinte_chrom_7.txt[1:3]
teosinte_chrom_7_hyphen.txt <- cbind(teosinte_chrom_7.txt_three_col, teosinte_chrom_7_hyphen.txt)
write.table(teosinte_chrom_7_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_7_hyphen.txt"))
```

```{r}
teosinte_chrom_8_hyphen.txt <- arrange(teosinte_chrom_8.txt, desc(Position))
ncol(teosinte_chrom_8_hyphen.txt)
teosinte_chrom_8_hyphen.txt <- apply(teosinte_chrom_8_hyphen.txt[,4:978], 2, function(teosinte_chrom_8_hyphen.txt) gsub("\\?", "-", teosinte_chrom_8_hyphen.txt))

teosinte_chrom_8.txt_three_col <- teosinte_chrom_8.txt[1:3]
teosinte_chrom_8_hyphen.txt <- cbind(teosinte_chrom_8.txt_three_col, teosinte_chrom_8_hyphen.txt)
write.table(teosinte_chrom_8_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_8_hyphen.txt"))
```

```{r}
teosinte_chrom_9_hyphen.txt <- arrange(teosinte_chrom_9.txt, desc(Position))
ncol(teosinte_chrom_9_hyphen.txt)
teosinte_chrom_9_hyphen.txt <- apply(teosinte_chrom_9_hyphen.txt[,4:978], 2, function(teosinte_chrom_9_hyphen.txt) gsub("\\?", "-", teosinte_chrom_9_hyphen.txt))

teosinte_chrom_9.txt_three_col <- teosinte_chrom_9.txt[1:3]
teosinte_chrom_9_hyphen.txt <- cbind(teosinte_chrom_9.txt_three_col, teosinte_chrom_9_hyphen.txt)
write.table(teosinte_chrom_9_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_9_hyphen.txt"))
```

```{r}
teosinte_chrom_10_hyphen.txt <- arrange(teosinte_chrom_10.txt, desc(Position))
ncol(teosinte_chrom_10_hyphen.txt)
teosinte_chrom_10_hyphen.txt <- apply(teosinte_chrom_10_hyphen.txt[,4:978], 2, function(teosinte_chrom_10_hyphen.txt) gsub("\\?", "-", teosinte_chrom_10_hyphen.txt))

teosinte_chrom_10.txt_three_col <- teosinte_chrom_10.txt[1:3]
teosinte_chrom_10_hyphen.txt <- cbind(teosinte_chrom_10.txt_three_col, teosinte_chrom_10_hyphen.txt)
write.table(teosinte_chrom_10_hyphen.txt, file = paste("./Teosinte_Files/teosinte_chrom_10_hyphen.txt"))
```

###Maize grep out 10 files, one for each chromosome
```{r}
maize_chrom_1.txt <- maize_ascending[grep("1$", maize_ascending$Chromosome), ]
write.table(maize_chrom_1.txt, file = paste("./Maize_Files/maize_chrom_1.txt"))
maize_chrom_2.txt <- maize_ascending[grep("2$", maize_ascending$Chromosome), ]
write.table(maize_chrom_2.txt, file = paste("./Maize_Files/maize_chrom_2.txt"))
maize_chrom_3.txt <- maize_ascending[grep("3$", maize_ascending$Chromosome), ]
write.table(maize_chrom_3.txt, file = paste("./Maize_Files/maize_chrom_3.txt"))
maize_chrom_4.txt <- maize_ascending[grep("4$", maize_ascending$Chromosome), ]
write.table(maize_chrom_4.txt, file = paste("./Maize_Files/maize_chrom_4.txt"))
maize_chrom_5.txt <- maize_ascending[grep("5$", maize_ascending$Chromosome), ]
write.table(maize_chrom_5.txt, file = paste("./Maize_Files/maize_chrom_5.txt"))
maize_chrom_6.txt <- maize_ascending[grep("6$", maize_ascending$Chromosome), ]
write.table(maize_chrom_6.txt, file = paste("./Maize_Files/maize_chrom_6.txt"))
maize_chrom_7.txt <- maize_ascending[grep("7$", maize_ascending$Chromosome), ]
write.table(maize_chrom_7.txt, file = paste("./Maize_Files/maize_chrom_7.txt"))
maize_chrom_8.txt <- maize_ascending[grep("8$", maize_ascending$Chromosome), ]
write.table(maize_chrom_8.txt, file = paste("./Maize_Files/maize_chrom_8.txt"))
maize_chrom_9.txt <- maize_ascending[grep("9$", maize_ascending$Chromosome), ]
write.table(maize_chrom_9.txt, file = paste("./Maize_Files/maize_chrom_9.txt"))
maize_chrom_10.txt <- maize_ascending[grep("10", maize_ascending$Chromosome), ]
write.table(maize_chrom_10.txt, file = paste("./Maize_Files/maize_chrom_10.txt"))

```


###Maize make 2 files, one with the 3 headers, and another to gsub out the "?" then cbind them together, to make 10 more maize files

```{r}
maize_chrom_1_hyphen.txt <- arrange(maize_chrom_1.txt, desc(Position))
ncol(maize_chrom_1_hyphen.txt)
maize_chrom_1_hyphen.txt <- apply(maize_chrom_1_hyphen.txt[,4:981], 2, function(maize_chrom_1_hyphen.txt) gsub("\\?", "-", maize_chrom_1_hyphen.txt))

maize_chrom_1.txt_three_col <- maize_chrom_1.txt[1:3]
maize_chrom_1_hyphen.txt <- cbind(maize_chrom_1.txt_three_col, maize_chrom_1_hyphen.txt)
write.table(maize_chrom_1_hyphen.txt, file = paste("./Maize_Files/maize_chrom_1_hyphen.txt"))
```


```{r}
maize_chrom_2_hyphen.txt <- arrange(maize_chrom_2.txt, desc(Position))
ncol(maize_chrom_2_hyphen.txt)
maize_chrom_2_hyphen.txt <- apply(maize_chrom_2_hyphen.txt[,4:981], 2, function(maize_chrom_2_hyphen.txt) gsub("\\?", "-", maize_chrom_2_hyphen.txt))

maize_chrom_2.txt_three_col <- maize_chrom_2.txt[1:3]
maize_chrom_2_hyphen.txt <- cbind(maize_chrom_2.txt_three_col, maize_chrom_2_hyphen.txt)
write.table(maize_chrom_2_hyphen.txt, file = paste("./Maize_Files/maize_chrom_2_hyphen.txt"))
```

```{r}
maize_chrom_3_hyphen.txt <- arrange(maize_chrom_3.txt, desc(Position))
ncol(maize_chrom_3_hyphen.txt)
maize_chrom_3_hyphen.txt <- apply(maize_chrom_3_hyphen.txt[,4:981], 2, function(maize_chrom_3_hyphen.txt) gsub("\\?", "-", maize_chrom_3_hyphen.txt))

maize_chrom_3.txt_three_col <- maize_chrom_3.txt[1:3]
maize_chrom_1_hyphen.txt <- cbind(maize_chrom_1.txt_three_col, maize_chrom_1_hyphen.txt)
write.table(maize_chrom_3_hyphen.txt, file = paste("./Maize_Files/maize_chrom_3_hyphen.txt"))
```

```{r}
maize_chrom_4_hyphen.txt <- arrange(maize_chrom_4.txt, desc(Position))
ncol(maize_chrom_4_hyphen.txt)
maize_chrom_4_hyphen.txt <- apply(maize_chrom_4_hyphen.txt[,4:981], 2, function(maize_chrom_4_hyphen.txt) gsub("\\?", "-", maize_chrom_4_hyphen.txt))

maize_chrom_4.txt_three_col <- maize_chrom_4.txt[1:3]
maize_chrom_4_hyphen.txt <- cbind(maize_chrom_4.txt_three_col, maize_chrom_4_hyphen.txt)
write.table(maize_chrom_4_hyphen.txt, file = paste("./Maize_Files/maize_chrom_4_hyphen.txt"))
```

```{r}
maize_chrom_5_hyphen.txt <- arrange(maize_chrom_5.txt, desc(Position))
ncol(maize_chrom_5_hyphen.txt)
maize_chrom_5_hyphen.txt <- apply(maize_chrom_5_hyphen.txt[,4:981], 2, function(maize_chrom_5_hyphen.txt) gsub("\\?", "-", maize_chrom_5_hyphen.txt))

maize_chrom_5.txt_three_col <- maize_chrom_5.txt[1:3]
maize_chrom_5_hyphen.txt <- cbind(maize_chrom_5.txt_three_col, maize_chrom_5_hyphen.txt)
write.table(maize_chrom_5_hyphen.txt, file = paste("./Maize_Files/maize_chrom_5_hyphen.txt"))
```

```{r}
maize_chrom_6_hyphen.txt <- arrange(maize_chrom_6.txt, desc(Position))
ncol(maize_chrom_6_hyphen.txt)
maize_chrom_6_hyphen.txt <- apply(maize_chrom_6_hyphen.txt[,4:981], 2, function(maize_chrom_6_hyphen.txt) gsub("\\?", "-", maize_chrom_6_hyphen.txt))

maize_chrom_6.txt_three_col <- maize_chrom_6.txt[1:3]
maize_chrom_6_hyphen.txt <- cbind(maize_chrom_6.txt_three_col, maize_chrom_6_hyphen.txt)
write.table(maize_chrom_6_hyphen.txt, file = paste("./Maize_Files/maize_chrom_6_hyphen.txt"))
```

```{r}
maize_chrom_7_hyphen.txt <- arrange(maize_chrom_7.txt, desc(Position))
ncol(maize_chrom_7_hyphen.txt)
maize_chrom_7_hyphen.txt <- apply(maize_chrom_7_hyphen.txt[,4:981], 2, function(maize_chrom_7_hyphen.txt) gsub("\\?", "-", maize_chrom_7_hyphen.txt))

maize_chrom_7.txt_three_col <- maize_chrom_7.txt[1:3]
maize_chrom_7_hyphen.txt <- cbind(maize_chrom_7.txt_three_col, maize_chrom_7_hyphen.txt)
write.table(maize_chrom_7_hyphen.txt, file = paste("./Maize_Files/maize_chrom_7_hyphen.txt"))
```

```{r}
maize_chrom_8_hyphen.txt <- arrange(maize_chrom_8.txt, desc(Position))
ncol(maize_chrom_8_hyphen.txt)
maize_chrom_8_hyphen.txt <- apply(maize_chrom_8_hyphen.txt[,4:981], 2, function(maize_chrom_8_hyphen.txt) gsub("\\?", "-", maize_chrom_8_hyphen.txt))

maize_chrom_8.txt_three_col <- maize_chrom_8.txt[1:3]
maize_chrom_8_hyphen.txt <- cbind(maize_chrom_8.txt_three_col, maize_chrom_8_hyphen.txt)
write.table(maize_chrom_8_hyphen.txt, file = paste("./Maize_Files/maize_chrom_8_hyphen.txt"))
```

```{r}
maize_chrom_9_hyphen.txt <- arrange(maize_chrom_9.txt, desc(Position))
ncol(maize_chrom_9_hyphen.txt)
maize_chrom_9_hyphen.txt <- apply(maize_chrom_9_hyphen.txt[,4:981], 2, function(maize_chrom_9_hyphen.txt) gsub("\\?", "-", maize_chrom_9_hyphen.txt))

maize_chrom_9.txt_three_col <- maize_chrom_9.txt[1:3]
maize_chrom_9_hyphen.txt <- cbind(maize_chrom_9.txt_three_col, maize_chrom_9_hyphen.txt)
write.table(maize_chrom_9_hyphen.txt, file = paste("./Maize_Files/maize_chrom_9_hyphen.txt"))
```

```{r}
maize_chrom_10_hyphen.txt <- arrange(maize_chrom_10.txt, desc(Position))
ncol(maize_chrom_10_hyphen.txt)
maize_chrom_10_hyphen.txt <- apply(maize_chrom_10_hyphen.txt[,4:981], 2, function(maize_chrom_10_hyphen.txt) gsub("\\?", "-", maize_chrom_10_hyphen.txt))

maize_chrom_10.txt_three_col <- maize_chrom_10.txt[1:3]
maize_chrom_10_hyphen.txt <- cbind(maize_chrom_10.txt_three_col, maize_chrom_10_hyphen.txt)
write.table(maize_chrom_10_hyphen.txt, file = paste("./Maize_Files/maize_chrom_10_hyphen.txt"))
```
#Data Visualization using ggplot
#Use pivot longer to neaten up data for ggplot input
##merge fang file with snp_position file

```{r}
fang_et_al_pivot_longer <- fang_et_al_genotypes %>% pivot_longer(!c(Sample_ID, JG_OTU, Group), names_to="SNP_ID", values_to= "NT")

snp_and_fang_et_al <- merge(fang_et_al_pivot_longer, snp_position, by="SNP_ID")

num_snp_and_fang_et_al <- (snp_and_fang_et_al[!is.na(as.numeric(snp_and_fang_et_al$Chromosome)),])
```

#Make first plot with ggplot, shows SNPs per chromosome using the newly merged file
```{r}
SNPs_per_chrom <- (num_snp_and_fang_et_al %>%
  select(SNP_ID, Chromosome, Position) %>%
  drop_na() %>%
  ggplot()+
  geom_bar(mapping = aes(as.numeric(Chromosome)), color="black", fill="blue") +
  labs(x = "Chromosome", y = "Total SNPs") +
  ggtitle("SNPs per Chromosome Position") +
  theme(plot.title = element_text(hjust = 0.5)))

pdf("Plots/SNPs_per_chrom.pdf")
print(SNPs_per_chrom)
```
#Make second plot with ggplot, shows SNP density per chromosome using the newly merged file for both maize and teosinte
```{r}
SNP_density <- (ggplot(num_snp_and_fang_et_al, aes(x= as.numeric(Position))) +
  geom_density(aes(fill = Chromosome)) +
  facet_wrap(~ Chromosome, nrow = 5, ncol = 2) +
  theme_bw() + labs(x = "Position", y = "Density") +
  ggtitle("SNP Density")+
  theme(plot.title = element_text(hjust = 0.5)))

pdf("Plots/SNPs_Density.pdf")
print(SNP_density)
```


##ggplot step
##Make a new column called Heterozygotes and then mark all missing rows and all Homozygous rows in the file
```{r}
hom_het_added_column <- num_snp_and_fang_et_al

hom_het_added_column$Heterozygotes <- "Heterozygotes"

hom_het_added_column$Heterozygotes[hom_het_added_column$NT == "?/?"] <- "Missing"

hom_het_added_column$Heterozygotes[hom_het_added_column$NT %in% c("A/A", "T/T", "C/C", "G/G")] <- "Homozygous"
```

#Make fourth plot with ggplot, uses the heteroyzgosity file from above to display the proportion of heterozygotes in both maize and teosinte
```{r}
Heterozygosity_of_All <- (ggplot(hom_het_added_column, aes(x = Sample_ID, fill = Heterozygotes)) +
  geom_bar(position = "fill") +
  labs(x = "Sample_ID", y = "Proportion") +
  ggtitle("Heterozygosity in Maize and Teosinte") +
  theme(plot.title = element_text(hjust = 0.5)))

pdf("Plots/Heterozygosity_of_All.pdf")
print(Heterozygosity_of_All)
```

#Make fifth plot with ggplot, shows Heterozygosity across all the groups using the newly merged file
```{r}
Heterozygosity_Across_Groups <- (ggplot(hom_het_added_column, aes(x = Group, fill = Heterozygotes)) +
  geom_bar(position = "fill") +
  labs(x = "Group", y = "Proportion") +
  theme(axis.text.x = element_text(angle = 90)) +
  ggtitle("Heterozygosity for all Groups") +
  theme(plot.title = element_text(hjust = 0.5)))

pdf("Plots/Heterozygosity_Across_Groups.pdf")
print(Heterozygosity_Across_Groups)

```

#Make sixth plot (my plot) with ggplot, I filtered out just Maize and then displayed the Heterozygosity across the three groups using the newly merged file
```{r}

hom_het_added_column_plot <- filter(hom_het_added_column, Group == "ZMMIL" | Group == "ZMMLR" | Group == "ZMMMR")

My_Plot_Heterozygosity_In_Maize <- (ggplot(hom_het_added_column_plot, aes(x = Group, fill = Heterozygotes)) +
  geom_bar(position = "fill") +
  labs(x = "Maize Group", y = "Proportion") +
  ggtitle("Heterozygosity in Just Maize") +
  theme(plot.title = element_text(hjust = 0.5)))

pdf("Plots/My_Plot_Heterozygosity_In_Maize.pdf")
print(My_Plot_Heterozygosity_In_Maize)

```
