---
title: "R_Markdown"
author: "Christopher_Zdyrski"
date: "3/13/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#This is my R Markdown document to analyze two snp files, combine them and pull out wanted data for further analysis in ggplot2. 

###Data Summary
###Get two original files

```{r}
fang_et_al_genotypes <- read.table('https://raw.githubusercontent.com/EEOB-BioData/BCB546-Spring2021/main/assignments/UNIX_Assignment/fang_et_al_genotypes.txt', header=TRUE, sep="\t",stringsAsFactors = FALSE)

snp_position <- read.table('https://raw.githubusercontent.com/EEOB-BioData/BCB546-Spring2021/main/assignments/UNIX_Assignment/snp_position.txt', header=TRUE, sep="\t", stringsAsFactors = FALSE)

```

###Attributes of fang_et_al_genotypes.txt

```{r}
dim(fang_et_al_genotypes)
str(fang_et_al_genotypes)
object.size(fang_et_al_genotypes)
```

#By inspecting this file I learned that:
#The file is 22681376 bytes.
#There are 2783 lines.
#There are 986 columns in this file.


###Attributes of snp_position.txt

```{r}
dim(snp_position)
str(snp_position)
object.size(snp_position)
```

#By inspecting this file I learned that:
#The file size is 327392 bytes.
#There are 983 lines.
#There are 15 columns in this file.



###Data processing

#Install & load packages
```{r}
install.packages("tidyverse")
library(tidyverse)

```


#Seperate out teosinte (ZMPBA, ZMPIL, and ZMPJA) and maize (ZMMIL OR ZMMLR OR ZMMMR) with grep, and also get rid of column two and three 

```{r}

teosinte_data <- fang_et_al_genotypes[grep("ZMP", fang_et_al_genotypes$Group), ]
maize_data <- fang_et_al_genotypes[grep("ZMM", fang_et_al_genotypes$Group), ]

teosinte_data_excluding_col <- select(teosinte_data, -JG_OTU, -Group)
teosinte_data_excluding_col <- as_tibble(teosinte_data_excluding_col)

maize_data_excluding_col <- select(maize_data, -JG_OTU, -Group)
maize_data_excluding_col <- as_tibble(maize_data_excluding_col)
```

#Swap col names to rows names

```{r}
swap_teosinte_data_excluding_col <- column_to_rownames(teosinte_data_excluding_col, var="Sample_ID")

swap_maize_data_excluding_col <- column_to_rownames(maize_data_excluding_col, var="Sample_ID")
```


#Transpose both genotype files to swap the rows and columns

```{r}
transposed_teosinte_data <- t(swap_teosinte_data_excluding_col)
transposed_maize_data <- t(swap_maize_data_excluding_col)

transposed_teosinte_data <- rownames_to_column(as.data.frame(transposed_teosinte_data), var = "SNP_ID")
transposed_maize_data <- rownames_to_column(as.data.frame(transposed_maize_data), var = "SNP_ID")
```


#Crop and Sort snp file
```{r}
snp_cropped <- select(snp_position, SNP_ID, Chromosome, Position)

attach(snp_cropped)
sorted_snp_position <- snp_cropped[order(SNP_ID),]
```

#Merge files (maize/teosinte with snp_position file)
```{r}
teosinte_joined <- merge(sorted_snp_position, transposed_teosinte_data, by="SNP_ID")

teosinte_joined_2 <- filter(teosinte_joined, Chromosome != "multiple" & Chromosome != "unknown")
teosinte_ascending <- arrange(teosinte_joined_2, Position)

maize_joined <- merge(sorted_snp_position, transposed_maize_data, by="SNP_ID")
maize_joined_2 <- filter(maize_joined, Chromosome != "multiple" & Chromosome != "unknown")

maize_ascending <- arrange(maize_joined_2, Position)

```


###Teosinte Awk out all the 20 files
```{r}
teosinte_chrom_1.txt <- teosinte_ascending[grep("1$", teosinte_ascending$Chromosome), ]
teosinte_chrom_2.txt <- teosinte_ascending[grep("2$", teosinte_ascending$Chromosome), ]
teosinte_chrom_3.txt <- teosinte_ascending[grep("3$", teosinte_ascending$Chromosome), ]
teosinte_chrom_4.txt <- teosinte_ascending[grep("4$", teosinte_ascending$Chromosome), ]
teosinte_chrom_5.txt <- teosinte_ascending[grep("5$", teosinte_ascending$Chromosome), ]
teosinte_chrom_6.txt <- teosinte_ascending[grep("6$", teosinte_ascending$Chromosome), ]
teosinte_chrom_7.txt <- teosinte_ascending[grep("7$", teosinte_ascending$Chromosome), ]
teosinte_chrom_8.txt <- teosinte_ascending[grep("8$", teosinte_ascending$Chromosome), ]
teosinte_chrom_9.txt <- teosinte_ascending[grep("9$", teosinte_ascending$Chromosome), ]
teosinte_chrom_10.txt <- teosinte_ascending[grep("10", teosinte_ascending$Chromosome), ]
```


###Teosinte make 2 files, one with the 3 headers, and another to gsub out the "?" then cbind them together
###Teosinte still have to sort in descending order

```{r}
teosinte_chrom_1_hyphen.txt <- arrange(teosinte_chrom_1.txt, desc(Position))
ncol(teosinte_chrom_1_hyphen.txt)
teosinte_chrom_1_hyphen.txt <- apply(teosinte_chrom_1_hyphen.txt[,4:978], 2, function(teosinte_chrom_1_hyphen.txt) gsub("\\?", "-", teosinte_chrom_1_hyphen.txt))

teosinte_chrom_1.txt_three_col <- teosinte_chrom_1.txt[1:3]
teosinte_chrom_1_hyphen.txt <- cbind(teosinte_chrom_1.txt_three_col, teosinte_chrom_1_hyphen.txt)
```

```{r}
teosinte_chrom_2_hyphen.txt <- arrange(teosinte_chrom_2.txt, desc(Position))
ncol(teosinte_chrom_2_hyphen.txt)
teosinte_chrom_2_hyphen.txt <- apply(teosinte_chrom_2_hyphen.txt[,4:978], 2, function(teosinte_chrom_2_hyphen.txt) gsub("\\?", "-", teosinte_chrom_2_hyphen.txt))

teosinte_chrom_2.txt_three_col <- teosinte_chrom_2.txt[1:3]
teosinte_chrom_2_hyphen.txt <- cbind(teosinte_chrom_2.txt_three_col, teosinte_chrom_2_hyphen.txt)
```

```{r}
teosinte_chrom_3_hyphen.txt <- arrange(teosinte_chrom_3.txt, desc(Position))
ncol(teosinte_chrom_3_hyphen.txt)
teosinte_chrom_3_hyphen.txt <- apply(teosinte_chrom_3_hyphen.txt[,4:978], 2, function(teosinte_chrom_3_hyphen.txt) gsub("\\?", "-", teosinte_chrom_3_hyphen.txt))

teosinte_chrom_3.txt_three_col <- teosinte_chrom_3.txt[1:3]
teosinte_chrom_3_hyphen.txt <- cbind(teosinte_chrom_3.txt_three_col, teosinte_chrom_3_hyphen.txt)
```

```{r}
teosinte_chrom_4_hyphen.txt <- arrange(teosinte_chrom_4.txt, desc(Position))
ncol(teosinte_chrom_4_hyphen.txt)
teosinte_chrom_4_hyphen.txt <- apply(teosinte_chrom_4_hyphen.txt[,4:978], 2, function(teosinte_chrom_4_hyphen.txt) gsub("\\?", "-", teosinte_chrom_4_hyphen.txt))

teosinte_chrom_4.txt_three_col <- teosinte_chrom_4.txt[1:3]
teosinte_chrom_4_hyphen.txt <- cbind(teosinte_chrom_4.txt_three_col, teosinte_chrom_4_hyphen.txt)
```

```{r}
teosinte_chrom_5_hyphen.txt <- arrange(teosinte_chrom_5.txt, desc(Position))
ncol(teosinte_chrom_5_hyphen.txt)
teosinte_chrom_5_hyphen.txt <- apply(teosinte_chrom_5_hyphen.txt[,4:978], 2, function(teosinte_chrom_5_hyphen.txt) gsub("\\?", "-", teosinte_chrom_5_hyphen.txt))

teosinte_chrom_5.txt_three_col <- teosinte_chrom_5.txt[1:3]
teosinte_chrom_5_hyphen.txt <- cbind(teosinte_chrom_5.txt_three_col, teosinte_chrom_5_hyphen.txt)
```

```{r}
teosinte_chrom_6_hyphen.txt <- arrange(teosinte_chrom_6.txt, desc(Position))
ncol(teosinte_chrom_6_hyphen.txt)
teosinte_chrom_6_hyphen.txt <- apply(teosinte_chrom_6_hyphen.txt[,4:978], 2, function(teosinte_chrom_6_hyphen.txt) gsub("\\?", "-", teosinte_chrom_6_hyphen.txt))

teosinte_chrom_6.txt_three_col <- teosinte_chrom_6.txt[1:3]
teosinte_chrom_6_hyphen.txt <- cbind(teosinte_chrom_6.txt_three_col, teosinte_chrom_6_hyphen.txt)
```

```{r}
teosinte_chrom_7_hyphen.txt <- arrange(teosinte_chrom_7.txt, desc(Position))
ncol(teosinte_chrom_7_hyphen.txt)
teosinte_chrom_7_hyphen.txt <- apply(teosinte_chrom_7_hyphen.txt[,4:978], 2, function(teosinte_chrom_7_hyphen.txt) gsub("\\?", "-", teosinte_chrom_7_hyphen.txt))

teosinte_chrom_7.txt_three_col <- teosinte_chrom_7.txt[1:3]
teosinte_chrom_7_hyphen.txt <- cbind(teosinte_chrom_7.txt_three_col, teosinte_chrom_7_hyphen.txt)
```

```{r}
teosinte_chrom_8_hyphen.txt <- arrange(teosinte_chrom_8.txt, desc(Position))
ncol(teosinte_chrom_8_hyphen.txt)
teosinte_chrom_8_hyphen.txt <- apply(teosinte_chrom_8_hyphen.txt[,4:978], 2, function(teosinte_chrom_8_hyphen.txt) gsub("\\?", "-", teosinte_chrom_8_hyphen.txt))

teosinte_chrom_8.txt_three_col <- teosinte_chrom_8.txt[1:3]
teosinte_chrom_8_hyphen.txt <- cbind(teosinte_chrom_8.txt_three_col, teosinte_chrom_8_hyphen.txt)
```

```{r}
teosinte_chrom_9_hyphen.txt <- arrange(teosinte_chrom_9.txt, desc(Position))
ncol(teosinte_chrom_9_hyphen.txt)
teosinte_chrom_9_hyphen.txt <- apply(teosinte_chrom_9_hyphen.txt[,4:978], 2, function(teosinte_chrom_9_hyphen.txt) gsub("\\?", "-", teosinte_chrom_9_hyphen.txt))

teosinte_chrom_9.txt_three_col <- teosinte_chrom_9.txt[1:3]
teosinte_chrom_9_hyphen.txt <- cbind(teosinte_chrom_9.txt_three_col, teosinte_chrom_9_hyphen.txt)
```

```{r}
teosinte_chrom_10_hyphen.txt <- arrange(teosinte_chrom_10.txt, desc(Position))
ncol(teosinte_chrom_10_hyphen.txt)
teosinte_chrom_10_hyphen.txt <- apply(teosinte_chrom_10_hyphen.txt[,4:978], 2, function(teosinte_chrom_10_hyphen.txt) gsub("\\?", "-", teosinte_chrom_10_hyphen.txt))

teosinte_chrom_10.txt_three_col <- teosinte_chrom_10.txt[1:3]
teosinte_chrom_10_hyphen.txt <- cbind(teosinte_chrom_10.txt_three_col, teosinte_chrom_10_hyphen.txt)
```

###Maize Awk out all the 20 files
```{r}
maize_chrom_1.txt <- maize_ascending[grep("1$", maize_ascending$Chromosome), ]
maize_chrom_2.txt <- maize_ascending[grep("2$", maize_ascending$Chromosome), ]
maize_chrom_3.txt <- maize_ascending[grep("3$", maize_ascending$Chromosome), ]
maize_chrom_4.txt <- maize_ascending[grep("4$", maize_ascending$Chromosome), ]
maize_chrom_5.txt <- maize_ascending[grep("5$", maize_ascending$Chromosome), ]
maize_chrom_6.txt <- maize_ascending[grep("6$", maize_ascending$Chromosome), ]
maize_chrom_7.txt <- maize_ascending[grep("7$", maize_ascending$Chromosome), ]
maize_chrom_8.txt <- maize_ascending[grep("8$", maize_ascending$Chromosome), ]
maize_chrom_9.txt <- maize_ascending[grep("9$", maize_ascending$Chromosome), ]
maize_chrom_10.txt <- maize_ascending[grep("10", maize_ascending$Chromosome), ]

```


###Maize make 2 files, one with the 3 headers, and another to gsub out the "?" then cbind them together
###Maize still have to sort in descending order

```{r}
maize_chrom_1_hyphen.txt <- arrange(maize_chrom_1.txt, desc(Position))
ncol(maize_chrom_1_hyphen.txt)
maize_chrom_1_hyphen.txt <- apply(maize_chrom_1_hyphen.txt[,4:981], 2, function(maize_chrom_1_hyphen.txt) gsub("\\?", "-", maize_chrom_1_hyphen.txt))

maize_chrom_1.txt_three_col <- maize_chrom_1.txt[1:3]
maize_chrom_1_hyphen.txt <- cbind(maize_chrom_1.txt_three_col, maize_chrom_1_hyphen.txt)
```


```{r}
maize_chrom_2_hyphen.txt <- arrange(maize_chrom_2.txt, desc(Position))
ncol(maize_chrom_2_hyphen.txt)
maize_chrom_2_hyphen.txt <- apply(maize_chrom_2_hyphen.txt[,4:981], 2, function(maize_chrom_2_hyphen.txt) gsub("\\?", "-", maize_chrom_2_hyphen.txt))

maize_chrom_2.txt_three_col <- maize_chrom_2.txt[1:3]
maize_chrom_2_hyphen.txt <- cbind(maize_chrom_2.txt_three_col, maize_chrom_2_hyphen.txt)
```

```{r}
maize_chrom_3_hyphen.txt <- arrange(maize_chrom_3.txt, desc(Position))
ncol(maize_chrom_3_hyphen.txt)
maize_chrom_3_hyphen.txt <- apply(maize_chrom_3_hyphen.txt[,4:981], 2, function(maize_chrom_3_hyphen.txt) gsub("\\?", "-", maize_chrom_3_hyphen.txt))

maize_chrom_3.txt_three_col <- maize_chrom_3.txt[1:3]
maize_chrom_1_hyphen.txt <- cbind(maize_chrom_1.txt_three_col, maize_chrom_1_hyphen.txt)
```

```{r}
maize_chrom_4_hyphen.txt <- arrange(maize_chrom_4.txt, desc(Position))
ncol(maize_chrom_4_hyphen.txt)
maize_chrom_4_hyphen.txt <- apply(maize_chrom_4_hyphen.txt[,4:981], 2, function(maize_chrom_4_hyphen.txt) gsub("\\?", "-", maize_chrom_4_hyphen.txt))

maize_chrom_4.txt_three_col <- maize_chrom_4.txt[1:3]
maize_chrom_4_hyphen.txt <- cbind(maize_chrom_4.txt_three_col, maize_chrom_4_hyphen.txt)
```

```{r}
maize_chrom_5_hyphen.txt <- arrange(maize_chrom_5.txt, desc(Position))
ncol(maize_chrom_5_hyphen.txt)
maize_chrom_5_hyphen.txt <- apply(maize_chrom_5_hyphen.txt[,4:981], 2, function(maize_chrom_5_hyphen.txt) gsub("\\?", "-", maize_chrom_5_hyphen.txt))

maize_chrom_5.txt_three_col <- maize_chrom_5.txt[1:3]
maize_chrom_5_hyphen.txt <- cbind(maize_chrom_5.txt_three_col, maize_chrom_5_hyphen.txt)
```

```{r}
maize_chrom_6_hyphen.txt <- arrange(maize_chrom_6.txt, desc(Position))
ncol(maize_chrom_6_hyphen.txt)
maize_chrom_6_hyphen.txt <- apply(maize_chrom_6_hyphen.txt[,4:981], 2, function(maize_chrom_6_hyphen.txt) gsub("\\?", "-", maize_chrom_6_hyphen.txt))

maize_chrom_6.txt_three_col <- maize_chrom_6.txt[1:3]
maize_chrom_6_hyphen.txt <- cbind(maize_chrom_6.txt_three_col, maize_chrom_6_hyphen.txt)
```

```{r}
maize_chrom_7_hyphen.txt <- arrange(maize_chrom_7.txt, desc(Position))
ncol(maize_chrom_7_hyphen.txt)
maize_chrom_7_hyphen.txt <- apply(maize_chrom_7_hyphen.txt[,4:981], 2, function(maize_chrom_7_hyphen.txt) gsub("\\?", "-", maize_chrom_7_hyphen.txt))

maize_chrom_7.txt_three_col <- maize_chrom_7.txt[1:3]
maize_chrom_7_hyphen.txt <- cbind(maize_chrom_7.txt_three_col, maize_chrom_7_hyphen.txt)
```

```{r}
maize_chrom_8_hyphen.txt <- arrange(maize_chrom_8.txt, desc(Position))
ncol(maize_chrom_8_hyphen.txt)
maize_chrom_8_hyphen.txt <- apply(maize_chrom_8_hyphen.txt[,4:981], 2, function(maize_chrom_8_hyphen.txt) gsub("\\?", "-", maize_chrom_8_hyphen.txt))

maize_chrom_8.txt_three_col <- maize_chrom_8.txt[1:3]
maize_chrom_8_hyphen.txt <- cbind(maize_chrom_8.txt_three_col, maize_chrom_8_hyphen.txt)
```

```{r}
maize_chrom_9_hyphen.txt <- arrange(maize_chrom_9.txt, desc(Position))
ncol(maize_chrom_9_hyphen.txt)
maize_chrom_9_hyphen.txt <- apply(maize_chrom_9_hyphen.txt[,4:981], 2, function(maize_chrom_9_hyphen.txt) gsub("\\?", "-", maize_chrom_9_hyphen.txt))

maize_chrom_9.txt_three_col <- maize_chrom_9.txt[1:3]
maize_chrom_9_hyphen.txt <- cbind(maize_chrom_9.txt_three_col, maize_chrom_9_hyphen.txt)
```

```{r}
maize_chrom_10_hyphen.txt <- arrange(maize_chrom_10.txt, desc(Position))
ncol(maize_chrom_10_hyphen.txt)
maize_chrom_10_hyphen.txt <- apply(maize_chrom_10_hyphen.txt[,4:981], 2, function(maize_chrom_10_hyphen.txt) gsub("\\?", "-", maize_chrom_10_hyphen.txt))

maize_chrom_10.txt_three_col <- maize_chrom_10.txt[1:3]
maize_chrom_10_hyphen.txt <- cbind(maize_chrom_10.txt_three_col, maize_chrom_10_hyphen.txt)
```
#Data Visualization using ggplot
#Use pivot longer to neaten up data for ggplot input
##chromosome of teosinte

```{r}
teosinte_joined_plot <- pivot_longer(teosinte_joined, Chromosome)

teosinte_joined_plot_numeric <- transform(teosinte_joined_plot, Chromosome=as.numeric(Chromosome), Position=as.numeric(Position))

ggplot(teosinte_joined_plot_numeric) +
  geom_point(aes(Chromosome, Position)) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  ggtitle("Teosinte chromosome plot") +
  theme(plot.title = element_text(hjust = 0.5))

```

##chromosome of maize

```{r}
maize_joined_plot <- pivot_longer(maize_joined, Chromosome)

maize_joined_plot_numeric <- transform(maize_joined_plot, Chromosome=as.numeric(Chromosome), Position=as.numeric(Position))

ggplot(maize_joined_plot_numeric) +
  geom_point(aes(Chromosome, Position)) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
   ggtitle("Maize chromosome plot") +
  theme(plot.title = element_text(hjust = 0.5))


```

##chromosome snp position

```{r}
snp_position %>%
  select(SNP_ID, Chromosome, Position) %>%
  drop_na() %>%
  ggplot() + 
  geom_bar(aes(x=Chromosome)) +
  ggtitle("SNPS per chromosome position") +
  theme(plot.title = element_text(hjust = 0.5))

```

##ggplot step
##homozygous and heterogygous file making

```{r}
homo_heto_teosinte_joined_plot_numeric <- teosinte_joined_plot_numeric %>% 
  mutate_if(is.character, str_replace_all, pattern = "A/A", replacement = "Homozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "T/T", replacement = "Homozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "C/C", replacement = "Homozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "G/G", replacement = "Homozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "A/T", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "A/C", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "A/G", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "T/C", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "T/G", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "T/A", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "C/T", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "C/G", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "C/C", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "G/T", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "G/C", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "G/A", replacement = "Heterozygous")

homo_heto_maize_joined_plot_numeric <- maize_joined_plot_numeric %>% 
  mutate_if(is.character, str_replace_all, pattern = "A/A", replacement = "Homozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "T/T", replacement = "Homozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "C/C", replacement = "Homozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "G/G", replacement = "Homozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "A/T", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "A/C", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "A/G", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "T/C", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "T/G", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "T/A", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "C/T", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "C/G", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "C/C", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "G/T", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "G/C", replacement = "Heterozygous") %>%
  mutate_if(is.character, str_replace_all, pattern = "G/A", replacement = "Heterozygous")

```

#Heterozygosity Plot of Maize and Teosinte
```{r}
homo_heto_teosinte_joined_plot_numeric %>%
  select(Position, gene) %>%
  ggplot() + 
  geom_point(aes(Position, gene)) +
  ggtitle("Heterozygosity in Maize and Teosinte") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank())
```

#Heterozygousity Plot of Teosinte
```{r}
homo_heto_teosinte_joined_plot_numeric %>%
  select(Position, gene) %>%
  ggplot() + 
  geom_point(aes(Position, gene)) +
  ggtitle("Heterozygosity in Teosinte") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) 
```

#Heterozygousity Plot of Maize
```{r}
homo_heto_maize_joined_plot_numeric %>%
  select(Position, gene) %>%
  ggplot() + 
  geom_point(aes(Position, gene)) +
  ggtitle("Heterozygosity in Maize") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) 
```

#Here is my plot looking at genes across SNPs
```{r}
snp_position %>%
  select(Position, gene) %>%
  ggplot() + 
  geom_point(aes(Position, gene)) +
  ggtitle("Which gene each SNP is on anaccoridng to its position") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) 

```
